libs-scala/resources-akka/src/test/suite/scala/com/digitalasset/resources/akka/AkkaResourceOwnerSpec.scala

```diff
@@ -104,4 +103,32 @@ final class AkkaResourceOwnerSpec extends AsyncWordSpec with Matchers {
       }
     }
   }
+
+  "a function returning a SourceQueue" should {
+    "convert to a ResourceOwner" in {
+      var number = 0
+      val resource = for {
+        actorSystem <- Factories
+          .forActorSystem(() => ActorSystem("TestActorSystem"))
+          .acquire()
+        materializer <- Factories.forMaterializer(() => Materializer(actorSystem)).acquire()
+        sourceQueue <- Factories
+          .forSourceQueue(
+            Source
+              .queue[Int](10, OverflowStrategy.backpressure)
+              .to(Sink.foreach(number = _))
+          )(materializer)
+          .acquire()
+      } yield sourceQueue
+
+      for {
+        queue <- resource.asFuture
+        offerResult <- queue.offer(123)
+        _ <- resource.release()
```

Comment:
This seems to be the pattern in this test file, but this is prone to leaks if tests fail. Do you think you can change the file to use `ResourceOwner#use` (in a separate PR)?

Reply:
Yes, I was thinking about this, will try.

Reply:
Sorry @SamirTalwar-DA, as there was some flakiness, I decided to put it into this PR. Please take a look.

Reply:
I reported the `lsp-tests` flakiness on Slack, btw.